tensorflow/compiler/xla/service/gpu/backend_configs.proto

```diff
@@ -88,3 +88,25 @@ message BitcastBackendConfig {
   LayoutProto source_layout = 1;
   LayoutProto result_layout = 2;
 }
+
+// Backend config for a fused Multi-Headed Attention (fMHA) that runs through
+// cudnn.
+message CudnnfMHABackendConfig {
+  // Opaque algorithm number and tuning knobs chosen for this fMHA.
+  stream_executor.dnn.AlgorithmProto algorithm = 8;
+
+  // The scaling factor multiplied with the BMM1 result. fmha_scale is 0 if the
+  // MHA pattern has no scaling.
+  double fmha_scale = 10;
+
+  // Dropout factor in MHA
+  double dropout_rate = 13;
+
+  xla.DotDimensionNumbers bmm1_dot_dimension_numbers = 11;
+  xla.DotDimensionNumbers bmm2_dot_dimension_numbers = 12;
+
+  xla.ShapeProto intermediate_tensor_shape = 14;
```

Comment:
Is it an arbitrary shape or is it forced by other factors?

Reply:
This is the intermediate tensor shape (i.e tensor shape between the BMMs) which can be technically inferred but we keep this for ease of use and to avoid any surprises with the layout.

Reply:
This shape could vary depending on the what kind of layout we take in for BMMs, i.e. QKV combined or KV combined. Inferring of this shape should technically already be done in the graph rewriter phase. We introduce this field here to pass the computed layout along to minimize shape-inferring complexity.

Reply:
Based on discussions on chat, resolving this.